apps/wallet-dashboard/app/(protected)/vesting/page.tsx

```diff
@@ -264,152 +256,175 @@ function VestingDashboardPage(): JSX.Element {
+                                }
+                            />
+
+                            <div className="flex flex-col px-lg py-sm">
+                                <div className="flex flex-row gap-md">
+                                    <DisplayStats
+                                        label="Your stake"
+                                        value={`${totalStakedFormatted} ${totalStakedSymbol}`}
+                                    />
+                                    <DisplayStats
+                                        label="Earned"
+                                        value={`${totalEarnedFormatted} ${totalEarnedSymbol}`}
+                                    />
+                                </div>
+                            </div>
+                            <div className="flex flex-col px-lg py-sm">
+                                <div className="flex w-full flex-col items-center justify-center space-y-4 pt-4">
+                                    {system &&
+                                        timelockedStakedObjectsGrouped?.map(
+                                            (timelockedStakedObject) => {
+                                                return (
+                                                    <StakedTimelockObject
+                                                        key={
+                                                            timelockedStakedObject.validatorAddress +
+                                                            timelockedStakedObject.stakeRequestEpoch +
+                                                            timelockedStakedObject.label
+                                                        }
+                                                        getValidatorByAddress={
+                                                            getValidatorByAddress
+                                                        }
+                                                        timelockedStakedObject={
+                                                            timelockedStakedObject
+                                                        }
+                                                        handleUnstake={handleUnstake}
+                                                        currentEpoch={Number(system.epoch)}
+                                                    />
+                                                );
+                                            },
+                                        )}
+                                </div>
+                            </div>
+                        </Panel>
+                    </div>
+                ) : null}
+
+                {isDialogStakeOpen && (
+                    <StakeDialog
+                        isTimelockedStaking
+                        stakedDetails={selectedStake}
+                        onSuccess={handleOnSuccess}
+                        isOpen
```

Comment:
This way the forms inside are actually emptied when the component unmounts and mounts again, in my opinion this is better instead of resetting states manually

Reply:
Should we remove the `isOpen` prop than?